mne/preprocessing/ica.py

```diff
@@ -531,11 +531,12 @@ def _transform_raw(self, raw, start, stop):
             raise RuntimeError('No fit available. Please fit ICA.')
         start, stop = _check_start_stop(raw, start, stop)
 
-        picks = [raw.ch_names.index(k) for k in self.ch_names]
+        picks = pick_types(raw.info, include=self.ch_names, exclude='bads',
+                           meg=False, ref_meg=False)
```

Comment:
there was some reason why we did not exclude bads here .... I only I could find the discussion ...